lib/src/front_end/piece_factory.dart

```diff
@@ -123,6 +133,69 @@ mixin PieceFactory implements CommentWriter {
     }
   }
 
+  // TODO(tall): Generalize this to work with if elements too.
```

Comment:
Why would this be useful? Do they share a representation in the analyzer?

Reply:
They don't have the same analyzer representation (alas), but they should have the same formatting logic. So what I'll do is write some sort of shim/conversion layer that lets me use the same IfPiece and (hopefully) same lowering code for both if statements and if elements.